inc/osdvar.h

```diff
@@ -138,8 +25,288 @@ typedef struct WAYPOINT_TYP {
 //    uint8_t autocontinue;
 } WAYPOINT, *WAYPOINT_PTR;
 
-extern WAYPOINT wp_list[MAX_WAYPOINTS];
+/* 
+
+osd_state_struct variables should have a lifetime that flows left to right
+like this:
+
+Serial reader (Mavlink/UAVTalk/Etc) => Airlock => OSDProc
+
+The Mavlink (mavlink_osd_state) and OSDProc (osdproc_osd_state) 
+instances of this struct are owned completely by Mavlink and
+OSDProc; no other threads should access them. The Airlock is
+where exchange between them happens, and access to the Airlock
+is controlled via a mutex. 
+
+There are two copy operations that use
+copy_osd_state_thread_safe to copy from Mavlink => Airlock, and
+Airlock => OSDProc.  The idea is to have simple, ready access to
+the osd_state_struct variable's contents during Mavlink reading 
+and OSD writing, and to avoid ever pending on a slow serial read
+operation during a fast screen drawing operation.
+
+If that were all there were to it, it would be very clean. But some
+code elsewhere (like Board.c) also updates values directly in the airlock,
+and the mutex is used to control access to these as well. 
+
+See also the other_osd_state_struct, which have a less predictable
+lifetime than osd_state_struct, and have a variety of convenience 
+accessors to make using them less aggravating and tedious.
+
+-- SLG
+*/
+typedef struct osd_state_struct osd_state;
+struct osd_state_struct {
```

Comment:
Should we rename all those members to not include the `osd_` prefix?
Since it is in the osd_state I think it is implied it is for the osd ;)

If the ide can rename struct fields it should be semi quick